packages/@lwc/engine-core/src/framework/mutation-tracker.ts

```diff
@@ -4,15 +4,40 @@
  * SPDX-License-Identifier: MIT
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
  */
-import { valueMutated, valueObserved } from '../libs/mutation-tracker';
+import {
+    JobFunction,
+    CallbackFunction,
+    ReactiveObserver,
+    valueMutated,
+    valueObserved,
+} from '../libs/mutation-tracker';
 import { VM } from './vm';
 
+const DUMMY_REACTIVE_OBSERVER = {
+    observe(job: JobFunction) {
+        job();
+    },
+    reset() {},
+    link() {},
+} as unknown as ReactiveObserver;
```

Comment:
I tried to do this more elegantly in TypeScript, but it ended up being a huge hassle because TypeScript complains about missing private properties on the `ReactiveObserver`. Doing this `unknown` cast results in a lot less code.

Reply:
The main concern that I have here is that the `DUMMY_REACTIVE_OBSERVER` is missing the `notify` method and TypeScript doesn't complain.

I think we can do better here without compromising on code size (yes, yes it's possible 😉). TypeScript complains because you are trying to tell TypeScript that `DUMMY_REACTIVE_OBSERVER` is an instance of `ReactiveObserver`. This is why it complains that private properties are missing.

The key here is to create an interface that would be implemented by both the `DUMMY_REACTIVE_OBSERVER` and the current `ReactiveObserver` class.  

```ts
export interface ReactiveObserver {
    observe(job: JobFunction): void;
    notify(): void;
    reset(): void;
    link(reactiveObservers: ReactiveObserver[]): void;
}

const DUMMY_REACTIVE_OBSERVER: ReactiveObserver = {  /* ... */ }

class ReactiveObserverImpl implements ReactiveObserver {}
```

Granted the `Impl` suffix really looks like Java, but it's the most straightforward way which guarantees type safety. The same observation applies to `AccessorReactiveObserver`.

Reply:
Great minds think alike! 🙂 This is exactly the code I started writing. But it was so much extra boilerplate that I wondered if it really added much value.

It especially got awkward because `ReactiveObserver` is imported from the file as both a class and a type, so consumers would have to know whether they were importing `ReactiveObserver` or `ReactiveObserverImpl`... it got messy fast.